### PR TITLE
hiredict 1.3.1 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

need a followup of adding conflicts_with

```
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /opt/homebrew
Could not symlink include/hiredis/adapters/ae.h
Target /opt/homebrew/include/hiredis/adapters/ae.h
```

### CODE REVIEW

File: Formula/h/hiredict.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  def install
+    system "make", "install", "PREFIX=#{prefix}", "USE_SSL=1"
+    pkgshare.install "examples"
+
+    # remove versioned dynamic library links
+    rm Dir["#{lib}/*.dylib.*"]
+  end
+
+  test do
+    # running `./test` requires a database to connect to, so just make
```
Comment (MikeMcQuaid) [inline Formula/h/hiredict.rb:20]: Would it be worth having a test dependency so we can test it connects?

File: Formula/h/hiredict.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  license all_of: ["BSD-3-Clause", "LGPL-3.0-or-later"]
+  head "https://codeberg.org/redict/hiredict.git", branch: "main"
+
+  depends_on "openssl@3"
+
+  def install
+    system "make", "install", "PREFIX=#{prefix}", "USE_SSL=1"
+    pkgshare.install "examples"
+
+    # remove versioned dynamic library links
```
Comment (MikeMcQuaid) [inline Formula/h/hiredict.rb:15]: Why?

File: Formula/h/hiredict.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    rm Dir["#{lib}/*.dylib.*"]
+  end
+
+  test do
+    # running `./test` requires a database to connect to, so just make
+    # sure it compiles
+    system ENV.cc, pkgshare/"examples/example.c", "-o", testpath/"test",
+                   "-I#{include}/hiredict", "-L#{lib}", "-lhiredict"
+    assert_predicate testpath/"test", :exist?
```
Comment (MikeMcQuaid) [inline Formula/h/hiredict.rb:24]:
```suggestion
```

as the test will fail if the above `system` fails

File: Formula/h/hiredict.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/h/hiredict.rb:15]: it caused some audit failure

```
hiredict
  * Non-libraries were installed to "/opt/homebrew/opt/hiredict/lib".
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /opt/homebrew/opt/hiredict/lib/libhiredict.dylib.1
      /opt/homebrew/opt/hiredict/lib/libhiredict_ssl.dylib.1.3.1
      /opt/homebrew/opt/hiredict/lib/libhiredict_ssl.dylib.1
```

File: Formula/h/hiredict.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/h/hiredict.rb:20]: I just did a copy from hiredis, I feel like the server starts does not worth the effort for the client lib stuff. wdyt?

File: Formula/h/hiredict.rb (hunk 2 above)
Comment (MikeMcQuaid) [reply Formula/h/hiredict.rb:15]: The file format is wrong. They should be `libhiredict_ssl.1.3.1.dylib`

Comment (ZhongRuoyu) [reply Formula/h/hiredict.rb:15]: Seems to be an upstream problem. Version suffixes for dylibs on macOS come before the `.dylib` extension name (`libhiredict.1.3.1.dylib`), which is different from other systems (`libhiredict.so.1.3.1` on Linux).

Working on a fix.

Comment (Bo98) [reply Formula/h/hiredict.rb:15]: Looks like an upstream bug. https://codeberg.org/redict/hiredict/src/commit/57291e2244ebb6c32784cd81244ba88d0e53b8b9/Makefile#L157 sets `DYLIB_MINOR_NAME` but they forgot to also set `DYLIB_MAJOR_NAME` similarly (like https://codeberg.org/redict/hiredict/src/commit/57291e2244ebb6c32784cd81244ba88d0e53b8b9/Makefile#L65).

Neither `SSL_DYLIB_MINOR_NAME` or `SSL_DYLIB_MAJOR_NAME` is set correctly either.

File: Formula/h/hiredict.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/h/hiredict.rb:20]: that is true, let me try out later today to see if I can do a full e2e test on it.

File: Formula/h/hiredict.rb (hunk 2 above)
Comment (ZhongRuoyu) [reply Formula/h/hiredict.rb:15]: Proposed fix: https://codeberg.org/redict/hiredict/commit/73e0f09f8ad89279ff116d5fb49ff2211c56f5f1.patch

Upstream PR: https://codeberg.org/redict/hiredict/pulls/16

Comment (chenrui333) [reply Formula/h/hiredict.rb:15]: Thanks @ZhongRuoyu! 💪

**Comment by chenrui333** [comment]
upstream still did not create a new tag per request, I will reopen this once they do that.

**Pull request closed without merging**